codegen/src/test/resources/software/amazon/awssdk/codegen/poet/batchmanager/test-batchmanager-async-class.java

```diff
@@ -0,0 +1,132 @@
+import software.amazon.awssdk.annotations.Generated;
+import software.amazon.awssdk.annotations.SdkInternalApi;
+import software.amazon.awssdk.core.batchmanager.BatchManager;
+import software.amazon.awssdk.core.batchmanager.BatchOverrideConfiguration;
+import software.amazon.awssdk.services.batchmanagertest.BatchManagerTestAsyncClient;
+import software.amazon.awssdk.services.batchmanagertest.batchmanager.BatchManagerTestAsyncBatchManager;
+import software.amazon.awssdk.services.batchmanagertest.model.DeleteRequestBatchResponse;
+import software.amazon.awssdk.services.batchmanagertest.model.DeleteRequestRequest;
+import software.amazon.awssdk.services.batchmanagertest.model.DeleteRequestResponse;
+import software.amazon.awssdk.services.batchmanagertest.model.SendRequestBatchResponse;
+import software.amazon.awssdk.services.batchmanagertest.model.SendRequestRequest;
+import software.amazon.awssdk.services.batchmanagertest.model.SendRequestResponse;
+
+@Generated("software.amazon.awssdk:codegen")
+@SdkInternalApi
+public final class DefaultBatchManagerTestAsyncBatchManager implements BatchManagerTestAsyncBatchManager {
+    private final BatchManagerTestAsyncClient client;
+
+    private final BatchManager<SendRequestRequest, SendRequestResponse, SendRequestBatchResponse> sendRequestBatchManager;
+
+    private final BatchManager<DeleteRequestRequest, DeleteRequestResponse, DeleteRequestBatchResponse> deleteRequestBatchManager;
+
+    private DefaultBatchManagerTestAsyncBatchManager(DefaultBuilder builder) {
+        this.client = builder.client;
+        ScheduledExecutorService scheduledExecutor = builder.scheduledExecutor;
+        this.sendRequestBatchManager = BatchManager
+            .builder(SendRequestRequest.class, SendRequestResponse.class, SendRequestBatchResponse.class)
+            .batchFunction(BatchManagerTestBatchFunctions.sendRequestBatchAsyncFunction(client))
+            .responseMapper(BatchManagerTestBatchFunctions.sendRequestResponseMapper())
+            .batchKeyMapper(BatchManagerTestBatchFunctions.sendRequestBatchKeyMapper())
+            .overrideConfiguration(sendRequestConfig(builder.overrideConfiguration)).scheduledExecutor(scheduledExecutor)
+            .build();
+        this.deleteRequestBatchManager = BatchManager
+            .builder(DeleteRequestRequest.class, DeleteRequestResponse.class, DeleteRequestBatchResponse.class)
+            .batchFunction(BatchManagerTestBatchFunctions.deleteRequestBatchAsyncFunction(client))
+            .responseMapper(BatchManagerTestBatchFunctions.deleteRequestResponseMapper())
+            .batchKeyMapper(BatchManagerTestBatchFunctions.deleteRequestBatchKeyMapper())
+            .overrideConfiguration(deleteRequestConfig(builder.overrideConfiguration)).scheduledExecutor(scheduledExecutor)
+            .build();
+    }
+
+    @SdkInternalApi
+    public DefaultBatchManagerTestAsyncBatchManager(BatchManagerTestAsyncClient client,
```

Comment:
Can we make it package private?

Reply:
Unit tests can't access this method if it is package private or protected.

Reply:
Can we move the unit tests to `software.amazon.awssdk.services.batchmanagertest.batchmanager.internal` package?

We should make it difficult for users to use internal APIs if possible.